docs/internationalization/core.md

```diff
@@ -159,6 +160,42 @@ The translation strings for device automations are defined under the `device_aut
 
 ```
 
+### Exceptions
+
+Localization is supported for `HomeAssistantError` and its subclasses.
+The translation strings for exceptions are defined under the `exception` key in a `strings.json` file. The example below describes the different supported keys.
+
+```json
+{
+  "exceptions": {
+    // Translations for known exceptions
+    "invalid_index": {
+      "message": "Invalid index selected, expected [0,1,2]. Got {index}"
+    }
+  }
+}
+
+```
+
+Example of raising an exception with localization during a service call:
+
+```python
+async def async_select_index(hass: HomeAssistant, index: int) -> None:
+    """Setup the config entry for my device."""
+    try:
+        check_index(index)
+    except ValueError as ex:
+        raise ServiceValidationError(
+            message,
```

Comment:
@jbouwh where does this variable come from.

Reply:
Good point. It should be a static string with a message to be logged.

Reply:
https://github.com/home-assistant/developers.home-assistant/pull/1991